toolchain/lower/testdata/function/generic/call_recursive_sccs_deep.carbon

```diff
@@ -0,0 +1,564 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/lower/testdata/function/generic/call_recursive_sccs_deep.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/lower/testdata/function/generic/call_recursive_sccs_deep.carbon
+
+import Core library "io";
+fn A[T:! type](x: T, count: i32) -> T;
+fn B[T:! type](x: T, count: i32);
+fn C[T:! type](x: T, count: i32);
+fn D[T:! type](x: T, count: i32) -> T;
+fn E[T:! type](x: T, count: i32) -> T;
+fn F[T:! type](x: T, count: i32) -> T;
+
+// Builds on `call_recursive_mutual.carbon` and call_recursive_diamond.carbon`
+// B-C form a mutually recursive SCC, D-E-F-G form a diamond SCC.
+// The two specifics for each of A, B, C, D, E, F, G with a ptr type could be deduplicated.
```

Comment:
(Probably should change "ptr" -> "pointer" in other comments too, I may have missed commenting on some.)
```suggestion
// The two specifics for each of A, B, C, D, E, F, G with a pointer type could be deduplicated.
```